compiler/x/codegen/DataSnippet.cpp

```diff
@@ -50,17 +51,27 @@ TR::IA32DataSnippet::addMetaDataForCodeAddress(uint8_t *cursor)
    // add dummy class unload/redefinition assumption.
    if (_isClassAddress)
       {
```

Comment:
this doesn't seem right to me. If there is supposedly a class address written at cursor, then why shouldn't we be creating some kind of `ArbitraryClassAddress` or `ClassObject` relocation? Typically you would create a relocation for the class pointer and that would take care of creating the class unload assumption at AOT load time.

There are other implementations of `addMetaDataForCodeAddress` sprinkled around the x86 code generator that do not require a new relocation record type to be created and that follow the model I outlined above.

Reply:
I think in this case the class address is "right", i.e. it was communicated to the compiler that generated the code by the current compiler that's loading the method and so the only thing necessary to relocate the address is to register an unload assumption against it. I'm not sure if that means we can reuse one of the existing relocations and just write have the loading compiler write the same value to the location or if that has other repercussions.

Reply:
So, after investigating the option of relocating the class pointer, I think it's not possible to
do from this data snippet, because the node, at which the relocation happens, is not related
to the class node. E.g.  older version of OpenJ9 used to create snippet for `instanceof` node.
 Thus we do not know anything about the class that will be stored in the snippet.

@dsouzai says that creating this kind of class pointer relocation will be possible with one of the future changes. So maybe this change should be postponed until then.